shimmingtoolbox/coils/coordinates.py

```diff
@@ -165,13 +167,12 @@ def resample_from_to(nii_from_img, nii_to_vox_map, order=2, mode='nearest', cval
 
     elif from_img.ndim == 4:
         nt = from_img.shape[3]
-        resampled_4d = np.zeros(nii_to_vox_map.shape + (nt,))
-
-        for it in range(nt):
-            nii_from_img_3d = nib.Nifti1Image(from_img[..., it], nii_from_img.affine, header=nii_from_img.header)
-            nii_resampled_3d = nib_resample_from_to(nii_from_img_3d, nii_to_vox_map, order=order, mode=mode,
-                                                    cval=cval, out_class=out_class)
-            resampled_4d[..., it] = nii_resampled_3d.get_fdata()
+        nb_cpu_used = int(mp.cpu_count())
+        results = Parallel(n_jobs=nb_cpu_used, backend='loky')(
+            delayed(_resample_4d)( nii_from_img, nii_to_vox_map, it, order, mode, cval, out_class)
+            for it in range(nt))
+        results.sort(key=lambda x: x[0])
```

Comment:
Are you sure you need "it" to sort the output data? Maybe joblib already outputs it in the right order.